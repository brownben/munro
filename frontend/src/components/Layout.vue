<template>
  <div class="w-10/12 mx-auto md:w-3/4 xl:w-2/3 md:mt-2">
    <h1
      v-if="title"
      class="my-2 text-3xl font-bold md:text-35xl font-heading text-main-700"
    >
      {{ title }}
    </h1>
    <div
      class="grid grid-cols-2 gap-8 mb-4 md:gap-6 lg:gap-8"
      :class="{ gapSmall: gapSmall }"
    >
      <slot />
    </div>
  </div>
</template>
<script>
export default {
  props: {
    title: { type: String, default: '' },
    gapSmall: { type: Boolean, default: false },
  },
}
</script>
<style lang="postcss" scoped>
.gapSmall {
  grid-gap: 1.25rem !important;
}
</style>
