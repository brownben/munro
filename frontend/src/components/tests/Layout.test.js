import { mount } from '@vue/test-utils'
import Layout from '@/components/Layout.vue'

test('Is a Vue Instance', () => {
  const wrapper = mount(Layout)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(Layout, {
    propsData: {
      title: 'Hello World!',
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
