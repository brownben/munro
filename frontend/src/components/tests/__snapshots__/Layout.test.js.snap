// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="w-10/12 mx-auto md:w-3/4 xl:w-2/3 md:mt-2"
>
  <h1
    class="my-2 text-3xl font-bold md:text-35xl font-heading text-main-700"
  >
    
    Hello World!
  
  </h1>
   
  <div
    class="grid grid-cols-2 gap-8 mb-4 md:gap-6 lg:gap-8"
  />
</div>
`;
